app/services/ValidatingPageBuilder.scala

```diff
@@ -182,21 +190,31 @@ class ValidatingPageBuilder @Inject() (pageBuilder: PageBuilder){
         }
     }
 
-  private def checkForSequencePageReuse(connections: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
-    val vertexMap = connections.map(pv => (pv.id, pv)).toMap
-    val mainFlow: List[String] = mainFlowPageIds(vertexMap)
-    val sequencePageIds = for{
-      pv <- connections.filterNot(_.flows.isEmpty)              // Sequences
+  private def checkAllFlowsHaveUniqueTerminationPage(vertices: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = vertices.map(pv => (pv.id, pv)).toMap
+    val mainFlow: List[String] = pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
+
+    vertices.filterNot(_.flows.isEmpty) // Sequences
+            .flatMap(_.flows).distinct  // Unique flow ids
+            .filter{id =>               // All those containing pages which dont link to an EndStanza
+              val pages = pageGraph(findPageIds(List(id)), vertexMap, mainFlow)
+              !pages.isEmpty && !pages.exists(_.endPage)
+            }
+            .map(MissingUniqueFlowTerminator(_))
+  }
+
+  private def checkForSequencePageReuse(vertices: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = vertices.map(pv => (pv.id, pv)).toMap
+    val mainFlow: List[String] = pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
+    val sequencePageIds: List[String] = for{
+      pv <- vertices.filterNot(_.flows.isEmpty)                 // Sequences
       flw <- pv.flows                                           // Flow Ids
       p <- pageGraph(findPageIds(List(flw)), vertexMap, mainFlow) // Pages below sequences
     } yield p.id
 
     sequencePageIds.groupBy(x => x).toList.collect{case m if m._2.length > 1 => PageOccursInMultiplSequenceFlows(m._1)}
   }
 
-  private def mainFlowPageIds(vertexMap: Map[String, PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[String] =
-    pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
-
   @tailrec
   // Given a list of stanza ids, find all connected pages (wont follow links)
   private def findPageIds(ids: List[String], seen: List[String] = Nil, acc: List[String] = Nil)(implicit stanzaMap: Map[String, Stanza]): List[String] =
```

Comment:
Maybe require a change to the method name as we now collect the ids of both page stanzas and end stanzas.

Reply:
removed the finding of endstanzas